test/unit/canvas_events.js

```diff
@@ -505,17 +503,33 @@
       });
       canvas.setActiveObject(activeSelection);
       setSubTargetCheckRecursive(activeSelection);
-      // perform MouseOver event on deepest nested subTarget
+
+      // perform MouseOver event on a deeply nested subTarget
+      var moveEvent = fabric.document.createEvent('HTMLEvents');
+      moveEvent.initEvent('mousemove', true, true);
+      var target = canvas.item(1);
+      canvas.targets = [
+        target.item(1),
+        target.item(1).item(1),
+        target.item(1).item(1).item(1)
+      ];
+      canvas._fireOverOutEvents(target, moveEvent);
       assert.equal(counterOver, 4, 'mouseover fabric event fired 4 times for primary hoveredTarget & subTargets');
-      assert.equal(canvas._hoveredTarget, activeSelection, 'activeSelection is _hoveredTarget');
+      assert.equal(canvas._hoveredTarget, target, 'activeSelection is _hoveredTarget');
       assert.equal(canvas._hoveredTargets.length, 3, '3 additional subTargets are captured as _hoveredTargets');
+
       // perform MouseOut even on all hoveredTargets
-      assert.equal(counterOver, 4, 'mouseout fabric event fired 4 times for primary hoveredTarget & subTargets');
+      canvas.targets = [];
+      canvas._fireOverOutEvents(null, moveEvent);
+      assert.equal(counterOut, 4, 'mouseout fabric event fired 4 times for primary hoveredTarget & subTargets');
       assert.equal(canvas._hoveredTarget, null, '_hoveredTarget has been set to null');
       assert.equal(canvas._hoveredTargets.length, 0, '_hoveredTargets array is empty');
     });
   });
 
+  // TODO: QUnit.test('mousemove: subTargetCheck: setCursorFromEvent considers subTargets')
+  // TODO: QUnit.test('mousemove: subTargetCheck: setCursorFromEvent considers subTargets in reverse order, so the top-most subTarget's .hoverCursor takes precedence')
+
```

Comment:
dunno if you saw this, but I did update this test so that it at least checks that the expected events fire for the subtargets